### PR TITLE
hardening/1724_AddTest_toCheck_V1id

### PR DESCRIPTION
**Pull request by raconte**
Check attribute,name and metadata

### CODE REVIEW

File: src/app/contextBroker/contextBroker.cpp [hunk 1]
```diff
@@ -267,6 +267,9 @@ bool            statNotifQueue;
 int             lsPeriod;
 bool            relogAlarms;
 bool            strictIdv1;
+bool 		metadataId;
+bool		attributeId;
+bool 		nameId;
```
Comment (kzangeli) [inline src/app/contextBroker/contextBroker.cpp]: indentation, the three lines

File: src/app/contextBroker/contextBroker.cpp [hunk 2]
```diff
@@ -319,7 +322,9 @@ bool            strictIdv1;
 #define LOG_SUMMARY_DESC       "log summary period in seconds (defaults to 0, meaning 'off')"
 #define RELOGALARMS_DESC       "log messages for existing alarms beyond the raising alarm log message itself"
 #define CHECK_v1_ID_DESC       "additional checks for id fields in the NGSIv1 API"
-
+#define CHECK_METADATA_V1      "additional checks for metadata  in the NGSIv1 API"
+#define CHECK_ATTRIBUTE_V1     "additional checks for attribute in the NGSIv1 API"
+#define CHECK_NAME_V1          "additional checks for name  in the NGSIv1 API"
```
Comment (kzangeli) [inline src/app/contextBroker/contextBroker.cpp]: double-spaces in CHECK_METADATA_V1 and CHECK_NAME_V1

File: src/app/contextBroker/contextBroker.cpp [hunk 3]
```diff
@@ -357,7 +362,7 @@ PaArgument paArgs[] =
   { "-ipv6",          &useOnlyIPv6,  "USEIPV6",        PaBool,   PaOpt, false,      false,  true,  USEIPV6_DESC       },
   { "-harakiri",      &harakiri,     "HARAKIRI",       PaBool,   PaHid, false,      false,  true,  HARAKIRI_DESC      },
 
-  { "-https",         &https,        "HTTPS",          PaBool,   PaOpt, false,      false,  true,  HTTPS_DESC         },
+  { "-https",         &https,        "HTTPSCHECK_NAME_V1",          PaBool,   PaOpt, false,      false,  true,  HTTPS_DESC         },
```
Comment (kzangeli) [inline src/app/contextBroker/contextBroker.cpp]: Is HTTPSCHECK_NAME_V1 and old definition?
I can't see it in the PR diff.

File: src/lib/common/globals.cpp [hunk 4]
```diff
@@ -53,6 +53,9 @@ bool                   semWaitStatistics    = false;
 bool                   timingStatistics     = false;
 bool                   notifQueueStatistics = false;
 bool                   checkIdv1            = false;
+bool                   checkMetadatav1      = false;
+bool                   checkAttributev1     = false;
+bool 		       checkNamev1          = false;
```
Comment (kzangeli) [inline src/lib/common/globals.cpp]: indentation

File: src/lib/common/globals.cpp [hunk 5]
```diff
@@ -136,7 +139,10 @@ void orionInit
   bool               _semWaitStatistics,
   bool               _timingStatistics,
   bool               _notifQueueStatistics,
-  bool _checkIdv1)
+  bool _checkIdv1,
+  bool _checkMetadatav1,
+  bool _checkAttributev1,
+  bool _checkNamev1)
 {
```
Comment (kzangeli) [inline src/lib/common/globals.cpp]: Indentation (_checkIdv1 etc aligned under _notifQueueStatistics)

File: src/lib/common/globals.cpp [hunk 6]
```diff
@@ -169,6 +175,9 @@ void orionInit
   strncpy(transactionId, "N/A", sizeof(transactionId));
 
   checkIdv1 = _checkIdv1;
+  checkMetadatav1 = _checkMetadatav1;
+  checkAttributev1 = _checkAttributev1;
+  checkNamev1 = _checkNamev1;
```
Comment (kzangeli) [inline src/lib/common/globals.cpp]: Indentation:

    checkIdv1         = _checkIdv1;
    checkMetadatav1   = _checkMetadatav1;
    checkAttributev1  = _checkAttributev1;
    checkNamev1       = _checkNamev1;
￼

**Comment by kzangeli** [comment]
Some functional tests will have to be modified also, as the arguments have changed

File: src/lib/common/globals.cpp (hunk 6 above)
Comment (raconte) [inline src/lib/common/globals.cpp]: Fixed in b28ecd1

File: src/lib/common/globals.cpp (hunk 5 above)
Comment (raconte) [inline src/lib/common/globals.cpp]: Fixed in b28ecd1

File: src/lib/common/globals.cpp (hunk 4 above)
Comment (raconte) [inline src/lib/common/globals.cpp]: Fixed in b28ecd1

File: src/app/contextBroker/contextBroker.cpp (hunk 3 above)
Comment (raconte) [inline src/app/contextBroker/contextBroker.cpp]: Fixed in b28ecd1

File: src/app/contextBroker/contextBroker.cpp (hunk 2 above)
Comment (raconte) [inline src/app/contextBroker/contextBroker.cpp]: Fixed in b28ecd1

File: src/app/contextBroker/contextBroker.cpp (hunk 1 above)
Comment (raconte) [inline src/app/contextBroker/contextBroker.cpp]: Fixed in b28ecd1

**Comment by fgalan** [comment]
Unfortunatelly, this PR is not actually addressing the #1724 topic. Note that #1724 is about new .test covering existing functionality, not about new functionality. Thus, this PR is going to be closed.

**Pull request closed without merging**